webapp/src/Twig/TwigExtension.php

```diff
@@ -167,6 +169,7 @@ public function getGlobals(): array
                                                $this->authorizationChecker->isGranted('ROLE_ADMIN') &&
                                                $this->config->get('data_source') === DOMJudgeService::DATA_SOURCE_CONFIGURATION_AND_LIVE_EXTERNAL,
             'doc_links'                     => $this->dj->getDocLinks(),
+            'allow_registration'            => $selfRegistrationCategoriesCount !== 0,
```

Comment:
Decided to put this in the Twig globals. This does mean one more query on one page, so do we want to do that? Or do we want to add this to all public pages separately? I'm not sure how/if we can do this query only on the public pages otherwise.

Reply:
So for every page under /public? Isn't it always needed for the menu for all pages?

Reply:
If you are logged in not. So that’s only public since you need to be logged in everywhere else.
But the query is also not expensive… alternative is we write the value to the symfony cache and update it when needed, but that’s more complex

Reply:
I'm not seeing what you mean. I read from the code that this will be executed on all pages in all sections, not only public pages?

Reply:
That's what I said what it currently does. I'm wondering if that's a problem and we should ONLY do it on public pages and if so, how.